src/Entity/BaseUser3.php

```diff
@@ -13,18 +13,10 @@
 
 namespace Sonata\UserBundle\Entity;
 
-use Sonata\UserBundle\Model\User as AbstractedUser;
-
-class BaseUser3 extends AbstractedUser
+/**
+ * This class is used to avoid error for users that have installed the version
+ * 5.12.0 of the SonataUserBundle and have not updated their User class.
+ */
+class BaseUser3 extends BaseUser
```

Comment:
The [BaseUser3.orm.xml](https://github.com/sonata-project/SonataUserBundle/pull/1685/files#diff-04c482ac96eadf5ce39b5f6dd08d04d6694842f96c5295076c94774a10add5c8) is not needed for this class ?

Reply:
Not anymore the Mapping listener is changing the type on during the ClassMetadata loading process. `array` type is working until DBAL 3 included, DBAL 4 removes the class ArrayType class.

Reply:
And the BaseUser definition will be used for BaseUser3 because the class extends the BaseUser class ?

Reply:
yes